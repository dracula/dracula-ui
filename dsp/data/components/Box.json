{
  "class": "component",
  "type": "page",
  "id": "Box",
  "name": "Box",
  "last_updated": "2021-03-14T00:08:30.329Z",
  "last_updated_by": "System",
  "description": "Box is the most primitive component of Dracula UI.\nUsing Box allows for consumers of the components library to compose\nmore complex patterns, components, and UIs.\n\nBox includes built-in Color and Spacing properties that make building\ncomplex components convenient and consistent.",
  "related_entity_ids": [],
  "tags": ["component"],
  "snippets": {
    "trigger": "box",
    "languages": {
      "javascriptreact": "<Box color=\"purple\">Dracula</Box>;\n",
      "html": "<div color=\"purple\" class=\"drac-box drac-bg-purple\">Dracula</div>\n"
    }
  },
  "ext_com_draculaui_variations": [
    {
      "title": "Usage",
      "html": "<div color=\"purple\" class=\"drac-box drac-bg-purple\">Dracula</div>\n",
      "react": "<Box color=\"purple\">Dracula</Box>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/Box.png",
      "svgData": "./dsp/assets/svgs/Box.svg"
    },
    {
      "title": "Colors",
      "html": "<div color=\"pinkPurple\" class=\"drac-box drac-bg-pink-purple\">Pink Purple</div>\n",
      "react": "<Box color=\"pinkPurple\">Pink Purple</Box>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxColors.png",
      "svgData": "./dsp/assets/svgs/BoxColors.svg"
    },
    {
      "title": "Rounded",
      "html": "<div style=\"display: flex; flex-direction: row\">\n  <div\n    color=\"purple\"\n    rounded=\"none\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-none\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"sm\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-sm\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"base\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"lg\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-lg\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"2xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-2xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"3xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-3xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"full\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-full\"\n  ></div>\n</div>\n",
      "react": "<div style={{ display: \"flex\", flexDirection: \"row\" }}>\n  <Box\n    color=\"purple\"\n    rounded=\"none\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"sm\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"base\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"lg\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"2xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"3xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"full\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxRounded.png",
      "svgData": "./dsp/assets/svgs/BoxRounded.svg"
    },
    {
      "title": "Generic Elements",
      "html": "<div>\n  <button as=\"button\" color=\"pinkPurple\" class=\"drac-box drac-bg-pink-purple\">\n    Button</button\n  ><a as=\"a\" color=\"green\" href=\"#\" class=\"drac-box drac-bg-green\">Link</a>\n</div>\n",
      "react": "<div>\n  <Box as=\"button\" color=\"pinkPurple\">\n    Button\n  </Box>\n  <Box as=\"a\" color=\"green\" href=\"#\">\n    Link\n  </Box>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxGeneric Elements.png",
      "svgData": "./dsp/assets/svgs/BoxGeneric Elements.svg"
    }
  ],
  "ext_com_draculaui_props": {
    "color": {
      "defaultValue": null,
      "description": "The background color.",
      "name": "color",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": {
        "name": "\"white\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
      }
    },
    "rounded": {
      "defaultValue": null,
      "description": "The border radius.",
      "name": "rounded",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"base\" | \"lg\" | \"xl\" | \"2xl\" | \"3xl\" | \"full\" | undefined"
      }
    },
    "spacing": {
      "defaultValue": null,
      "description": "Dracula UI standard spacing properties.",
      "name": "spacing",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": { "name": "SpacingPropType | undefined" }
    },
    "margin": {
      "defaultValue": null,
      "description": "Dracula UI standard margin properties.",
      "name": "margin",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": { "name": "MarginPropType | undefined" }
    },
    "as": {
      "defaultValue": null,
      "description": "The HTML element to be used",
      "name": "as",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": { "name": "keyof HTMLElementTagNameMap | undefined" }
    }
  },
  "ext_com_draculaui_docgen": {
    "tags": {},
    "description": "Box is the most primitive component of Dracula UI.\nUsing Box allows for consumers of the components library to compose\nmore complex patterns, components, and UIs.\n\nBox includes built-in Color and Spacing properties that make building\ncomplex components convenient and consistent.",
    "displayName": "Box",
    "methods": [],
    "props": {
      "color": {
        "defaultValue": null,
        "description": "The background color.",
        "name": "color",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": {
          "name": "\"white\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
        }
      },
      "rounded": {
        "defaultValue": null,
        "description": "The border radius.",
        "name": "rounded",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"base\" | \"lg\" | \"xl\" | \"2xl\" | \"3xl\" | \"full\" | undefined"
        }
      },
      "spacing": {
        "defaultValue": null,
        "description": "Dracula UI standard spacing properties.",
        "name": "spacing",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": { "name": "SpacingPropType | undefined" }
      },
      "margin": {
        "defaultValue": null,
        "description": "Dracula UI standard margin properties.",
        "name": "margin",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": { "name": "MarginPropType | undefined" }
      },
      "as": {
        "defaultValue": null,
        "description": "The HTML element to be used",
        "name": "as",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": { "name": "keyof HTMLElementTagNameMap | undefined" }
      }
    }
  }
}
