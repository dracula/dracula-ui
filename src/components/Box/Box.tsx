import cx from 'classnames/dedupe'
import React, { AllHTMLAttributes } from 'react'
import { borderColors, colors, glowColors } from '../../base/colors'
import {
  PaddingMixin,
  paddingMixin,
  marginMixin,
  MarginMixin
} from '../../base/spacing'

type Element = HTMLElementTagNameMap

export const roundedBorders = {
  none: 'drac-rounded-none',
  sm: 'drac-rounded-sm',
  base: 'drac-rounded',
  lg: 'drac-rounded-lg',
  xl: 'drac-rounded-xl',
  '2xl': 'drac-rounded-2xl',
  '3xl': 'drac-rounded-3xl',
  full: 'drac-rounded-full'
}

/**
 * Box Props
 */
export type BoxProps<K extends keyof Element = 'div'> = {
  /** The background color. */
  color?: keyof typeof colors

  /** The glow color. */
  glowColor?: keyof typeof glowColors

  /** The border color. */
  borderColor?: keyof typeof borderColors

  /** The border radius. */
  rounded?: keyof typeof roundedBorders

  /** The HTML element to be used */
  as?: K
} & AllHTMLAttributes<K> &
  PaddingMixin &
  MarginMixin

/**
 * Box is the most primitive component of Dracula UI.
 * Using Box allows for consumers of the components library to compose
 * more complex patterns, components, and UIs.
 *
 * Box includes built-in Color and Spacing properties that make building
 * complex components convenient and consistent.
 */
export function Box<T extends keyof Element>(props: BoxProps<T>) {
  const finalProps = {
    ...props,
    className: cx(
      `drac-box`,
      props.className,
      props.color && colors[props.color],
      props.glowColor && glowColors[props.glowColor],
      props.borderColor && borderColors[props.borderColor],
      props.rounded && roundedBorders[props.rounded],
      ...paddingMixin(props),
      ...marginMixin(props)
    )
  }

  const as = finalProps.as ?? 'div'
  return React.createElement(as, finalProps, props.children)
}

Box.displayName = 'Box'
