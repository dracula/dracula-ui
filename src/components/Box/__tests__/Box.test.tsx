import React from 'react'
import * as ReactDOM from 'react-dom'
import { Box } from '@/components/Box/Box'

describe('Box', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Box color="white" />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})
