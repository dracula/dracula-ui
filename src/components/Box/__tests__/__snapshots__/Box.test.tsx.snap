// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site: Box Basic Usage 1`] = `
Object {
  "docs": "A Box is analogous to an HTML div and can contain any other component. Use Box when you want to compose layouts, or multiple components.",
  "html": "<div class=\\"drac-box\\">
  <span class=\\"drac-text drac-line-height drac-text-white\\">Dracula</span>
</div>
",
  "react": "<Box>
  <Text>Dracula</Text>
</Box>;
",
  "title": "Usage",
}
`;

exports[`Site: Box Colors 1`] = `
Object {
  "docs": "Boxes can be customized to use any of the Dracula UI theme colors, including all solid, gradient, and animated colors.",
  "html": "<div class=\\"drac-box drac-bg-pink-purple\\">
  <span class=\\"drac-text drac-line-height drac-text-black\\">Pink Purple</span>
</div>
",
  "react": "<Box color=\\"pinkPurple\\">
  <Text color=\\"black\\">Pink Purple</Text>
</Box>;
",
  "title": "Colors",
}
`;

exports[`Site: Box Display 1`] = `
Object {
  "docs": "Control the display type of an element.",
  "html": "<div class=\\"drac-box\\">
  <div
    class=\\"drac-box drac-bg-purple-cyan drac-d-inline-block drac-rounded-lg drac-p-xs drac-mr-xs\\"
  >
    <span class=\\"drac-text drac-line-height drac-text-black\\">1</span>
  </div>
  <div
    class=\\"drac-box drac-bg-purple-cyan drac-d-inline-block drac-rounded-lg drac-p-xs drac-mr-xs\\"
  >
    <span class=\\"drac-text drac-line-height drac-text-black\\">2</span>
  </div>
  <div
    class=\\"drac-box drac-bg-purple-cyan drac-d-inline-block drac-rounded-lg drac-p-xs drac-mr-xs\\"
  >
    <span class=\\"drac-text drac-line-height drac-text-black\\">3</span>
  </div>
</div>
",
  "react": "<Box>
  <Box color=\\"purpleCyan\\" display=\\"inline-block\\" rounded=\\"lg\\" p=\\"xs\\" mr=\\"xs\\">
    <Text color=\\"black\\">1</Text>
  </Box>
  <Box color=\\"purpleCyan\\" display=\\"inline-block\\" rounded=\\"lg\\" p=\\"xs\\" mr=\\"xs\\">
    <Text color=\\"black\\">2</Text>
  </Box>
  <Box color=\\"purpleCyan\\" display=\\"inline-block\\" rounded=\\"lg\\" p=\\"xs\\" mr=\\"xs\\">
    <Text color=\\"black\\">3</Text>
  </Box>
</Box>;
",
  "title": "Display",
}
`;

exports[`Site: Box Height 1`] = `
Object {
  "docs": "Set an element with a fixed height, including \`auto\` and \`full\`.",
  "html": "<div style=\\"display: flex; align-items: flex-end\\" class=\\"drac-box\\">
  <div
    class=\\"drac-box drac-h-xxs drac-bg-cyan-green drac-rounded-lg drac-p-xs drac-mr-xs\\"
  >
    <span class=\\"drac-text drac-line-height drac-text-black\\">xxs</span>
  </div>
  <div
    class=\\"drac-box drac-h-xs drac-bg-cyan-green drac-rounded-lg drac-p-xs drac-mr-xs\\"
  >
    <span class=\\"drac-text drac-line-height drac-text-black\\">xs</span>
  </div>
  <div
    class=\\"drac-box drac-h-sm drac-bg-cyan-green drac-rounded-lg drac-p-xs drac-mr-xs\\"
  >
    <span class=\\"drac-text drac-line-height drac-text-black\\">sm</span>
  </div>
  <div
    class=\\"drac-box drac-h-md drac-bg-cyan-green drac-rounded-lg drac-p-xs drac-mr-xs\\"
  >
    <span class=\\"drac-text drac-line-height drac-text-black\\">md</span>
  </div>
</div>
",
  "react": "<Box style={{ display: \\"flex\\", alignItems: \\"flex-end\\" }}>
  <Box color=\\"cyanGreen\\" height=\\"xxs\\" rounded=\\"lg\\" p=\\"xs\\" mr=\\"xs\\">
    <Text color=\\"black\\">xxs</Text>
  </Box>
  <Box color=\\"cyanGreen\\" height=\\"xs\\" rounded=\\"lg\\" p=\\"xs\\" mr=\\"xs\\">
    <Text color=\\"black\\">xs</Text>
  </Box>
  <Box color=\\"cyanGreen\\" height=\\"sm\\" rounded=\\"lg\\" p=\\"xs\\" mr=\\"xs\\">
    <Text color=\\"black\\">sm</Text>
  </Box>
  <Box color=\\"cyanGreen\\" height=\\"md\\" rounded=\\"lg\\" p=\\"xs\\" mr=\\"xs\\">
    <Text color=\\"black\\">md</Text>
  </Box>
</Box>;
",
  "title": "Height",
}
`;

exports[`Site: Box Rounded 1`] = `
Object {
  "docs": "Choose from one of the predefined \`rounded\` options for a Box.",
  "html": "<div class=\\"drac-box\\">
  <div class=\\"drac-box drac-bg-purple-cyan drac-rounded-lg drac-p-md\\">
    <span class=\\"drac-text drac-line-height drac-text-black\\"
      >Large round border</span
    >
  </div>
  <div class=\\"drac-box drac-bg-yellow-pink drac-rounded-sm drac-p-md\\">
    <span class=\\"drac-text drac-line-height drac-text-black\\"
      >small round border</span
    >
  </div>
</div>
",
  "react": "<Box>
  <Box rounded=\\"lg\\" color=\\"purpleCyan\\" p=\\"md\\">
    <Text color=\\"black\\">Large round border</Text>
  </Box>
  <Box rounded=\\"sm\\" color=\\"yellowPink\\" p=\\"md\\">
    <Text color=\\"black\\">small round border</Text>
  </Box>
</Box>;
",
  "title": "Rounded",
}
`;

exports[`Site: Box Scrollbar 1`] = `
Object {
  "docs": "Boxes come with primitive scrollbars, which allow you to customize the colors and shapes.",
  "html": "<div>
  <div class=\\"drac-box drac-scrollbar-purple drac-p-xs\\">
    <span class=\\"drac-text drac-line-height drac-text-black\\">xs padding</span>
  </div>
  <div class=\\"drac-box drac-scrollbar-pink drac-p-xs\\">
    <span class=\\"drac-text drac-line-height drac-text-black\\">xs padding</span>
  </div>
</div>
",
  "react": "<div>
  <Box scrollbar={true} p=\\"xs\\">
    <Text color=\\"black\\">xs padding</Text>
  </Box>
  <Box scrollbar=\\"pink\\" p=\\"xs\\">
    <Text color=\\"black\\">xs padding</Text>
  </Box>
</div>;
",
  "title": "Scrollbar",
}
`;

exports[`Site: Box Spacing 1`] = `
Object {
  "docs": "Boxes come with pre defined spacing primitives for padding and margin that allow for customizing the spacing inside and around boxes.",
  "html": "<div class=\\"drac-box\\">
  <div class=\\"drac-box drac-bg-yellow-pink drac-p-xs\\">
    <span class=\\"drac-text drac-line-height drac-text-black\\">xs padding</span>
  </div>
  <div class=\\"drac-box drac-bg-purple-cyan drac-p-sm\\">
    <span class=\\"drac-text drac-line-height drac-text-black\\">sm padding</span>
  </div>
  <div class=\\"drac-box drac-bg-pink-purple drac-p-md\\">
    <span class=\\"drac-text drac-line-height drac-text-black\\"
      >medium padding</span
    >
  </div>
  <div class=\\"drac-box drac-bg-yellow-pink drac-m-xs\\">
    <span class=\\"drac-text drac-line-height drac-text-black\\">xs margin</span>
  </div>
  <div class=\\"drac-box drac-bg-purple-cyan drac-m-sm\\">
    <span class=\\"drac-text drac-line-height drac-text-black\\">sm margin</span>
  </div>
  <div class=\\"drac-box drac-bg-pink-purple drac-m-md\\">
    <span class=\\"drac-text drac-line-height drac-text-black\\"
      >medium margin</span
    >
  </div>
</div>
",
  "react": "<Box>
  <Box color=\\"yellowPink\\" p=\\"xs\\">
    <Text color=\\"black\\">xs padding</Text>
  </Box>
  <Box color=\\"purpleCyan\\" p=\\"sm\\">
    <Text color=\\"black\\">sm padding</Text>
  </Box>
  <Box color=\\"pinkPurple\\" p=\\"md\\">
    <Text color=\\"black\\">medium padding</Text>
  </Box>
  <Box color=\\"yellowPink\\" m=\\"xs\\">
    <Text color=\\"black\\">xs margin</Text>
  </Box>
  <Box color=\\"purpleCyan\\" m=\\"sm\\">
    <Text color=\\"black\\">sm margin</Text>
  </Box>
  <Box color=\\"pinkPurple\\" m=\\"md\\">
    <Text color=\\"black\\">medium margin</Text>
  </Box>
</Box>;
",
  "title": "Spacing",
}
`;

exports[`Site: Box Width 1`] = `
Object {
  "docs": "Set an element with a fixed width, including \`auto\` and \`full\`.",
  "html": "<div class=\\"drac-box\\">
  <div
    class=\\"drac-box drac-w-xs drac-bg-yellow-pink drac-rounded-lg drac-p-xs drac-mb-xs\\"
  >
    <span class=\\"drac-text drac-line-height drac-text-black\\">xs</span>
  </div>
  <div
    class=\\"drac-box drac-w-md drac-bg-yellow-pink drac-rounded-lg drac-p-xs drac-mb-xs\\"
  >
    <span class=\\"drac-text drac-line-height drac-text-black\\">md</span>
  </div>
  <div
    class=\\"drac-box drac-w-xl drac-bg-yellow-pink drac-rounded-lg drac-p-xs drac-mb-xs\\"
  >
    <span class=\\"drac-text drac-line-height drac-text-black\\">xl</span>
  </div>
  <div
    class=\\"drac-box drac-w-full drac-bg-yellow-pink drac-rounded-lg drac-p-xs drac-mb-xs\\"
  >
    <span class=\\"drac-text drac-line-height drac-text-black\\">full</span>
  </div>
</div>
",
  "react": "<Box>
  <Box color=\\"yellowPink\\" width=\\"xs\\" rounded=\\"lg\\" p=\\"xs\\" mb=\\"xs\\">
    <Text color=\\"black\\">xs</Text>
  </Box>
  <Box color=\\"yellowPink\\" width=\\"md\\" rounded=\\"lg\\" p=\\"xs\\" mb=\\"xs\\">
    <Text color=\\"black\\">md</Text>
  </Box>
  <Box color=\\"yellowPink\\" width=\\"xl\\" rounded=\\"lg\\" p=\\"xs\\" mb=\\"xs\\">
    <Text color=\\"black\\">xl</Text>
  </Box>
  <Box color=\\"yellowPink\\" width=\\"full\\" rounded=\\"lg\\" p=\\"xs\\" mb=\\"xs\\">
    <Text color=\\"black\\">full</Text>
  </Box>
</Box>;
",
  "title": "Width",
}
`;
